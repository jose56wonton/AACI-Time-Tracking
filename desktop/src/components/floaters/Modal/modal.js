import React, { Component } from 'react';

import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import MUModal from '@material-ui/core/Modal';

import style from './styles';

export class Modal extends Component {
  render() {
    const { classes } = this.props;

    return (
      <MUModal open={this.props.open} onClose={this.props.toggle}>
        <div className={classes.paper}>{this.props.children}</div>
      </MUModal>
    );
  }
}

Modal.propTypes = {
  classes: PropTypes.object.isRequired,
  open: PropTypes.bool.isRequired,
  toggle: PropTypes.func.isRequired,
  children: PropTypes.array,
};

export default withStyles(style)(Modal);