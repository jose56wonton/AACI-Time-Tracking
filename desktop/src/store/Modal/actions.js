import { modalActionTypes } from 'constants/ActionTypes';

export const openModal = (type, props) => {
  return {
    type: modalActionTypes.SHOW_MODAL,
    modalType: type,
    modalProps: {
      ...props
    }
  };
};
export const closeModal = type => {
  return {
    type: modalActionTypes.HIDE_MODAL,
    modalType: type
  };
};