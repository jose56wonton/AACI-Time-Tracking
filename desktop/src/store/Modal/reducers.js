import { modalActionTypes } from 'constants/ActionTypes';

export const modalInitialState = {
  modalType: null,
  modalProps: {}
};

export default (state = modalInitialState, action) => {
  switch (action.type) {
  case modalActionTypes.SHOW_MODAL:
    return {
      modalType: action.modalType,
      modalProps: action.modalProps
    };
  case modalActionTypes.HIDE_MODAL:
    return modalInitialState;
  default:
    return state;
  }
};