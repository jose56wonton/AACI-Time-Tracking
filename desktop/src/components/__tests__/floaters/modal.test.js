import React from 'react';
import { shallow } from 'enzyme';

import { Modal } from 'components/floaters/Modal/modal';
import ModalHOC from 'components/floaters/Modal';

const props = {
  open: true,
  toggle: jest.fn(),
  children: <div>child</div>,
  classes: {}
};

const setup = () => {  
  return shallow(<Modal {...props} />);    
};

const setupHOC = () => {
  return shallow(<ModalHOC {...props} />);
};

describe('Modal Components', () => {
  it('should render correctly', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();    
  });
  it('should render correctly withStyles', () => {
    const wrapper = setupHOC();
    expect(wrapper).toMatchSnapshot();
  });
});