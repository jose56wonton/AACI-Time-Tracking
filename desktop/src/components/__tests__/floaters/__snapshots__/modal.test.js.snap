// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Component should render correctly 1`] = `
<WithStyles(Modal)
  onClose={[Function]}
  open={true}
>
  <div>
    <div>
      child
    </div>
  </div>
</WithStyles(Modal)>
`;

exports[`Modal Component should render correctly withStyles 1`] = `
<Modal
  classes={
    Object {
      "paper": "Modal-paper-1",
    }
  }
  open={true}
  toggle={[Function]}
>
  <div>
    child
  </div>
</Modal>
`;
